client/my-sites/exporter/index.jsx

```diff
@@ -35,6 +37,10 @@ function mapStateToProps( state ) {
 		didComplete: getExportingState( state, siteId ) === States.COMPLETE,
 		didFail: getExportingState( state, siteId ) === States.FAILED,
 		showGuidedTransferOptions: config.isEnabled( 'manage/export/guided-transfer' ),
+		purchaseGuidedTransfer: () => {
+			const siteSlug = getSiteSlug( state, siteId );
+			page( `/settings/export/${siteSlug}/guided` );
+		},
```

Comment:
The page navigation could be decoupled from the Redux logic if we pass just `siteSlug` as the prop, then do `page( `/settings/export/${this.props.siteSlug}/guided` );` inside the component.